core/src/main/java/io/jenkins/pluginhealth/scoring/probes/AbstractJavaImportsCheckProbe.java

```diff
@@ -0,0 +1,93 @@
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        final Path scmRepository = context.getScmRepository();
+
+        /* The "maxDepth" is set to Integer.MAX_VALUE because a repository can have multiple modules with class files in it. We do not want to miss any ".java" file.  */
+        try (Stream<Path> paths = Files.find(scmRepository, Integer.MAX_VALUE, (path, $) ->
+            Files.isRegularFile(path) && path.getFileName().toString().endsWith(".java"))) {
+            Set<String> javaFilesWithDeprecatedImports = paths.map(this::getFileNameWithImports)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+
+            return javaFilesWithDeprecatedImports.isEmpty()
+                ? ProbeResult.success(key(), String.format(getSuccessMessage() + " at %s plugin.", plugin.getName()))
+                : ProbeResult.failure(key(), String.format(getFailureMessage() + " at %s plugin for ", plugin.getName()) + javaFilesWithDeprecatedImports.stream().sorted(Comparator.naturalOrder()).collect(Collectors.joining(", ")));
+        } catch (IOException ex) {
+            LOGGER.error("Could not browse the plugin folder during {} probe. {}", key(), ex);
+            return ProbeResult.error(key(), String.format("Could not browse the plugin folder during {} probe.", plugin.getName()));
+        }
+    }
+
+    @Override
+    public String[] getProbeResultRequirement() {
+        return new String[]{SCMLinkValidationProbe.KEY};
+    }
+
+    /**
+     * Looks for all the imports in a path that should be checked.
+     *
+     * @param path The file Path to check imports.
+     * @return name of the file that contains the required imports.
+     */
+    private String getFileNameWithImports(Path path) {
+        try (Stream<String> fileContent = Files.lines(path).filter(line -> line.startsWith("import")).map(this::getFullyQualifiedImportName)) {
+            if (CollectionUtils.containsAny(fileContent.toList(), getListOfImports())) {
+                return path.getFileName().toString();
+            }
+        } catch (IOException ex) {
+            LOGGER.error("Could not browse the {} plugin folder during probe. {}", key(), ex);
+        }
+        return null;
+    }
+
+    /**
+     * @return a success message.
+     */
+    abstract String getSuccessMessage();
+
+    /**
+     * @return a failure message.
+     */
+    abstract String getFailureMessage();
+
+    /**
+     * Gets a fully qualified name of the imported class/library from an {@code import} statement.
+     *
+     * @param importStatement the statement from which the fully qualified name should be fetched.
+     * @return a String that contains only the fully qualified name of the class/library.
+     */
+    private String getFullyQualifiedImportName(String importStatement) {
+        return importStatement.replace("import ", "").replace(";", "");
```

Comment:
You can add a `trim()` to be sure that there are no particular spaces remaining around the statement that you want.